libsolidity/ast/AST.h

```diff
@@ -803,6 +805,7 @@ class FunctionDefinition: public CallableDeclaration, public StructurallyDocumen
 	bool isConstructor() const { return m_kind == Token::Constructor; }
 	bool isFallback() const { return m_kind == Token::Fallback; }
 	bool isReceive() const { return m_kind == Token::Receive; }
+	bool isFree() const { return m_free; }
```

Comment:
Btw where is this "free functions" term coming from? C++?

I think `isFree` looks kind of weird to be honest for anyone not 100% familiar with this.

Reply:
I would call it `free`, but you cannot do that in C++. I'm happy to use a totally different term.

Reply:
I think it is a C++ term, yes: https://en.wikipedia.org/wiki/Free_function